zebra/rt_netlink.c

```diff
@@ -483,7 +483,9 @@ static int netlink_route_change_read_unicast(struct nlmsghdr *h, ns_id_t ns_id,
 				memcpy(&nh.gate, gate, sz);
 
 			if (index) {
-				ifp = if_lookup_by_index(index, VRF_UNKNOWN);
+				ifp = vrf_lookup_interface_by_index(
+					    zebra_ns_lookup(ns_id)->if_table,
+					    index);
 				if (ifp)
```

Comment:
I think you should use `if_lookup_by_index_per_ns()` here instead (please see comment below).

Reply:
done